database/src/main/kotlin/no/nav/su/se/bakover/database/DatabaseEx.kt

```diff
@@ -9,36 +9,63 @@ import java.sql.Array
 import java.util.UUID
 import javax.sql.DataSource
 
+private fun sjekkUgyldigParameternavn(params: Map<String, Any?>) {
+    require(params.keys.none { it.contains(Regex("[æÆøØåÅ]")) }) { "Parameter-mapping forstår ikke særnorske tegn" }
+}
+
 internal fun String.oppdatering(
     params: Map<String, Any?>,
     session: Session
-) = session.run(queryOf(statement = this, paramMap = params).asUpdate)
+) {
+    sjekkUgyldigParameternavn(params)
+    session.run(queryOf(statement = this, paramMap = params).asUpdate)
+}
 
 internal fun <T> String.hent(
     params: Map<String, Any> = emptyMap(),
     session: Session,
     rowMapping: (Row) -> T
-): T? =
-    session.run(queryOf(this, params).map { row -> rowMapping(row) }.asSingle)
+): T? {
+    sjekkUgyldigParameternavn(params)
+    return session.run(queryOf(this, params).map { row -> rowMapping(row) }.asSingle)
+}
 
 internal fun <T> String.hentListe(
     params: Map<String, Any> = emptyMap(),
     session: Session,
     rowMapping: (Row) -> T
-): List<T> = session.run(queryOf(this, params).map { row -> rowMapping(row) }.asList)
+): List<T> {
+    sjekkUgyldigParameternavn(params)
+    return session.run(queryOf(this, params).map { row -> rowMapping(row) }.asList)
+}
 
 internal fun Row.uuid(name: String) = UUID.fromString(string(name))
 internal fun Row.uuid30(name: String) = UUID30.fromString(string(name))
+internal fun Row.uuid30OrNull(name: String) = stringOrNull(name)?.let { UUID30.fromString(it) }
 internal fun Row.tidspunkt(name: String) = this.instant(name).toTidspunkt()
 
 internal fun Session.inClauseWith(values: List<String>): Array =
     this.connection.underlying.createArrayOf("text", values.toTypedArray())
 
+internal fun <T> DataSource.withSession(session: Session?, block: (session: Session) -> T): T =
```

Comment:
Ville det blitt enklere å bruke dersom `session: Session? = null`

Reply:
Nja, det blir fortsatt litt knot siden andre argument sendes inn som en lambda.
Så jeg tror det vi har nå med eksplisitt overload er det smootheste.

Reply:
Aha, jeg stusset bare på at session var nullable. Burde man kalle overloaden istedenfor å sende session=null? Eller, det var kanskje for å være bakoverkompatible.